drivers/hmis/lib/hmis_data_cleanup/util.rb

```diff
@@ -27,43 +37,104 @@ def self.clear_enrollment_export_ids!
     # we are dealing with a single-CoC installation at the time of writing this.
     def self.update_all_enrollment_cocs!(coc_code)
       without_papertrail_or_timestamps do
-        Hmis::Hud::Enrollment.hmis.update_all(EnrollmentCoC: coc_code)
+        rows_affected = Hmis::Hud::Enrollment.hmis.where.not(EnrollmentCoC: coc_code).update_all(EnrollmentCoC: coc_code)
+        Rails.logger.info "Updated EnrollmentCoC on #{rows_affected} enrollments"
       end
     end
 
     # Assign Household ID where missing
-    #
-    # Note: If this gets to be a large number, an upsert is probably worth doing.
-    # We could `scope.find_in_batches` and then enrollment.assign_attributes and finally "import" the batch with a conflict key of id.
     def self.assign_missing_household_ids!
-      scope = Hmis::Hud::Enrollment.hmis.where(household_id: nil)
-      Rails.logger.info "Assigning household id to #{scope.size} enrollments"
-      scope.each do |enrollment|
-        hh_id = Hmis::Hud::Base.generate_uuid
-        without_papertrail_or_timestamps do
-          enrollment.update_columns(household_id: hh_id) # skips callbacks
+      Hmis::Hud::Enrollment.hmis.where(household_id: [nil, '']).find_in_batches do |batch|
```

Comment:
Assuming we can just use any random UUID for the initial household id, it seems like this could all be done in the server. This might be preferable if we are trying to make the operation as performant as possible. Maybe something like this?

```ruby
sql = 'Update "Enrollment" SET "HouseholdID" = gen_random_uuid() WHERE "HouseholdID" IS NULL OR "HouseholdID" = \'\''
result = Hmis::Hud::Enrollment.connection.raw_connection.exec(sql)
# result is PG::Result 
Rails.logger.info "Assigned HouseholdID to #{result.cmd_tuples} Enrollments"
```

Reply:
I liked @eanders suggestion of making it consistent based on the enrollment ID, because then if we need to run it repeatedly (say, the nightly imports are still going ) then it generates consistent IDs. Not that that really matters, I don't think, but it seems appropriate. I don't think performance on this one is huge because it should be a relatively smaller subset of enrollments.

Reply:
I _think_ we're trying to be consistent with the new importer extension, which uses the MD5 pattern below.  We could probably figure out how to do that in the DB, but I'm not sure it's worth it.